src/v/cluster/rm_stm.cc

```diff
@@ -2611,6 +2614,42 @@ ss::future<> rm_stm::handle_eviction() {
       });
 }
 
+void rm_stm::spawn_background_clean_for_pids(rm_stm::clear_type type) {
+    switch (type) {
+    case rm_stm::clear_type::tx_pids: {
+        if (_log_state.fence_pid_epoch.size() <= _max_saved_pids_count()) {
+            return;
+        }
+        break;
+    }
+    case rm_stm::clear_type::idempotent_pids: {
+        if (_log_state.seq_table.size() <= _max_saved_pids_count()) {
```

Comment:
Here we compare either `seq_table.size()` or `fence_pid_epoch.size()` with `_max_saved_pids_count`. While `clear_old_idempotent_pids` checks the size of  `idempotent_pids_order`. Is that intended ?